### PR TITLE
Polio-667: vaccine form a changes

### PR DESCRIPTION
**Pull request by hakifran**
## Self proof reading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] Did I add translations
- [x] My migrations file are included
- [ ] Are there enough tests

## Changes
- Add forma_date field dateInput component
- Add forma_comment TextArea field
- Add forma_date and forma_comment migrations

## How to test
- Edit a campaign
- Vaccine management tab
- Form A section
- Edit form A date and form A comment

## Print screen / video
Peek 2022-11-04 13-37

### CODE REVIEW

File: plugins/polio/js/src/forms/ScopeForm.tsx [hunk 1]
```diff
@@ -17,6 +17,36 @@ type Round = {
 type Values = {
     separate_scopes_per_round?: boolean;
     rounds: Round[];
+    initial_org_unit: number;
+};
+
+type Scope = {
```
Comment (quang-le) [inline plugins/polio/js/src/forms/ScopeForm.tsx]: This doesn't look like it's part of this PR

File: plugins/polio/js/src/hooks/useFormValidator.js [hunk 2]
```diff
@@ -486,6 +490,11 @@ const useRoundShape = () => {
             .typeError(formatMessage(MESSAGES.invalidDate))
             .nullable()
             .hasAllFormAFieldsDate(formatMessage),
+        forma_date: yup
+            .date()
+            .typeError(formatMessage(MESSAGES.invalidDate))
+            .nullable(),
```
Comment (quang-le) [inline plugins/polio/js/src/hooks/useFormValidator.js]: you need to use the same `yup` method as for other date fields for Form A

File: plugins/polio/js/src/pages/Budget/CreateBudgetStep/CreateBudgetStep.tsx [hunk 3]
```diff
@@ -220,7 +220,7 @@ const CreateBudgetStep: FunctionComponent<Props> = ({
                 dataTestId="Test-modal"
                 onClose={() => null}
             >
-                {userHasTeam && (
```
Comment (quang-le) [inline plugins/polio/js/src/pages/Budget/CreateBudgetStep/CreateBudgetStep.tsx:223]: This also looks like it's part of another PR (maybe on my side)

**Review by quang-le — CHANGES_REQUESTED** [review]
Nice work, just a few details:

Form validation is missing on the last date field

Screenshot 2022-11-07 at 10 36 07

Please checkout the code from `ScopeInput` which doesn't seem to be part of the PR. We don't want to ship breaking/incomplete code by mistake. I will check the budegt one on my end

**Review by beygorghor — CHANGES_REQUESTED** [review]
Screenshot 2022-11-07 at 11 00 47
- don't forget to add margin top on storage status modal
- also i have an infinite loop while trying to display a textarea

**Comment by quang-le** [comment]
> Screenshot 2022-11-07 at 11 00 47
> - don't forget to add margin top on storage status modal - also i have an infinite loop while trying to display a textarea

For the infinite loop, we should merge POLIO-670. I solved it there

**Comment by hakifran** [comment]
> > Screenshot 2022-11-07 at 11 00 47
> > - don't forget to add margin top on storage status modal - also i have an infinite loop while trying to display a textarea
> 
> For the infinite loop, we should merge POLIO-670. I solved it there
Screenshot from 2022-11-07 17-18-10
I solved the margin problem

File: plugins/polio/js/src/pages/Budget/CreateBudgetStep/CreateBudgetStep.tsx (hunk 3 above)
Comment (hakifran) [reply plugins/polio/js/src/pages/Budget/CreateBudgetStep/CreateBudgetStep.tsx:223]: Solved

File: plugins/polio/js/src/hooks/useFormValidator.js (hunk 2 above)
Comment (hakifran) [reply plugins/polio/js/src/hooks/useFormValidator.js]: solved

File: plugins/polio/js/src/forms/ScopeForm.tsx (hunk 1 above)
Comment (hakifran) [reply plugins/polio/js/src/forms/ScopeForm.tsx]: solved

**Comment by codeKameleon** [comment]
I spotted a bug with the comments : 
[recording-2022-11-07-17-17-33.webm](https://user-images.githubusercontent.com/25134301/200509919-f60e95f7-b639-4274-a582-ecd8db4cbf45.webm)

**Review by codeKameleon — CHANGES_REQUESTED** [review]
I spotted a bug with the comment  : 
[recording-2022-11-07-17-17-33.webm](https://user-images.githubusercontent.com/25134301/200509919-f60e95f7-b639-4274-a582-ecd8db4cbf45.webm)

**Comment by quang-le** [comment]
> I spotted a bug with the comment : [recording-2022-11-07-17-17-33.webm](https://user-images.githubusercontent.com/25134301/200509919-f60e95f7-b639-4274-a582-ecd8db4cbf45.webm)

I fixed it in POLIO-670, which merges into this branch

@hakifran : FYI

File: plugins/polio/js/src/hooks/useFormValidator.js [hunk 4]
```diff
@@ -256,6 +260,39 @@ yup.addMethod(
         });
     },
 );
+yup.addMethod(
```
Comment (quang-le) [inline plugins/polio/js/src/hooks/useFormValidator.js:263]: This is removed in POLIO-670. FYI

File: hat/assets/js/apps/Iaso/domains/storages/components/StatusModal.tsx [hunk 5]
```diff
@@ -104,13 +104,16 @@ const StatusModal: FunctionComponent<Props> = ({
...
                     />
-                    <TextArea
-                        label={formatMessage(MESSAGES.comment)}
-                        value={status?.comment}
-                        onChange={newComment =>
-                            handleChange('comment', newComment)
-                        }
-                    />
+
+                    <Box sx={{ mt: 2 }}>
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/storages/components/StatusModal.tsx:108]: This is changed to `<Box mt={2}>` in POLIO-670, FYI

**Review by quang-le — APPROVED** [review]
LGTM

**Pull request merged**